features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/site/themes/wso2/templates/stats/api-usage-by-applications/js/stats.js

```diff
@@ -0,0 +1,250 @@
+            apiName: apiName,
+            apiVersion: apiVersion,
+            fromDate: fromDate,
+            toDate: toDate,
+            apiFilter: apiFilter
+        },
+        function (json) {
+            $('#spinner').hide();
+            if (!json.error) {
+                var length = json.usage.length;
+                $('#noData').empty();
+                $('div#apiUsageByApplicationsTable_wrapper.dataTables_wrapper.no-footer').remove();
+
+                var $dataTable = $('<table class="display table table-striped table-bordered" width="100%" cellspacing="0" id="apiUsageByApplicationsTable"></table>');
+
+                $dataTable.append($('<thead class="tableHead"><tr>' +
+                    '<th>API</th>' +
+                    '<th>VERSION</th>' +
+                    '<th>APPLICATION NAME</th>' +
+                    '<th>USAGE</th>' +
+                    '</tr></thead>'));
+
+                for (var i = 0; i < json.usage.length; i++) {
+                    $dataTable.append($('<tr><td>' + json.usage[i].values.apiName + '</td><td>' + json.usage[i].values.apiVersion + '</td><td>' + json.usage[i].values.applicationName + '</td><td>' + json.usage[i].values.requstCount + '</td></tr>'));
+                }
+                if (length == 0) {
+                    $('#apiUsageByApplicationsTable').hide();
+                    $('div#apiUsageByApplicationsTable_wrapper.dataTables_wrapper.no-footer').remove();
+                    $('#noData').html('');
+                    $('#noData').append($('<div class="center-wrapper"><div class="col-sm-4"/><div class="col-sm-4 message message-info"><h4><i class="icon fw fw-info" title="No Stats"></i>' + i18n.t("No Data Available") + '</h4></div></div>'));
+
+                } else {
+                    $('#tableContainer').append($dataTable);
+                    $('#tableContainer').show();
+                    $('#apiUsageByApplicationsTable').datatables_extended({
+                        "order": [[3, "desc"]],
+                        "fnDrawCallback": function () {
+                            if (this.fnSettings().fnRecordsDisplay() <= $("#apiUsageByApplicationsTable_length option:selected").val()
+                                || $("#apiUsageByApplicationsTable_length option:selected").val() == -1)
+                                $('#apiUsageByApplicationsTable_paginate').hide();
+                            else
+                                $('#apiUsageByApplicationsTable_paginate').show();
+                        },
+                    });
+                }
+            } else {
+                if (json.message == "AuthenticateError") {
+                    jagg.showLogin();
+                } else {
+                    jagg.message({content: json.message, type: "error"});
+                }
+            }
+        }, "json");
+}
+
+function getDateTime(currentDay, fromDay) {
+    to = convertTimeString(currentDay);
+    from = convertTimeString(fromDay);
```

Comment:
I don't think this is implemented to work with different timezones (browser in a different timezone and the servers are in a different one). Please check this. See https://github.com/wso2/carbon-apimgt/pull/5758/files#diff-c6b01b5ecf5cd966a32cf31947e85592R148 a sample code